packages/core/content-releases/server/src/services/release.ts

```diff
@@ -452,27 +452,42 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
     /**
      * We separate publish and unpublish actions group by content type
      * And we keep only their ids to fetch them later to get all information needed
+     *
+     * We also need to separate collectionType entries from singleType entries because they work differently
      */
-    const actions: {
+    const collectionTypeActions: {
       [key: UID.ContentType]: {
         entriestoPublishIds: ReleaseAction['entry']['id'][];
         entriesToUnpublishIds: ReleaseAction['entry']['id'][];
       };
     } = {};
+    const singleTypeActions: {
+      uid: UID.ContentType;
+      id: ReleaseAction['entry']['id'];
+      action: ReleaseAction['type'];
+    }[] = [];
```

Comment:
what is the difference between publishing a single type and a collection type? what prevented from using the same logic?

Reply:
I found 2 problems when trying to publish singleType with the collectionType:

1. `findMany`: I can use findMany to get all the entries from one collectionType, but not for singleType, because `entityService.findMany` [if the entry is a singleType is doing a findOne ](https://github.com/strapi/strapi/blob/eb64ae5745bd7585cc21514afbf2d49f39961d94/packages/core/strapi/src/services/entity-service/index.ts#L135) so is not possible to get multiple singleTypes with different locales.
2. `publishMany`: This is actually the same problem than before. publishMany is using `entityService.findMany`([see](https://github.com/strapi/strapi/blob/30ae2308e85bb8612689ca9e12c784384cad22f0/packages/core/content-manager/server/src/services/entity-manager.ts#L286)) so if you pass singleType entries, it try to use findMany and happens the same error

Reply:
> findMany: I can use findMany to get all the entries from one collectionType, but not for singleType, because entityService.findMany [if the entry is a singleType is doing a findOne ](https://github.com/strapi/strapi/blob/eb64ae5745bd7585cc21514afbf2d49f39961d94/packages/core/strapi/src/services/entity-service/index.ts#L135)so is not possible to get multiple singleTypes with different locales.

I see :) 

Probably not worth the refactor (now), but I would use the database layer for those things. And specially more in v5, where the entity service will be deleted, and the strapi.documents.findMany will not return many locales at once for collection types either.